lab_1_classify_by_unigrams/main.py

```diff
@@ -6,106 +6,193 @@
+
+
+def calculate_frequencies(tokens):
+    if not isinstance(tokens, list):
+        return None
+    if isinstance(tokens, list):
+        for element in tokens:
+            if not isinstance(element, str):
+                return None
+    dictionary = {}
+    token_number = len(tokens)
+    for letter in tokens:
+        if letter not in dictionary.keys():
+            dictionary[letter] = 1/token_number
+        else:
+            dictionary[letter] = (dictionary[letter]*token_number + 1)/token_number
+    #print(dictionary)
+    return dictionary
+
+
+def create_language_profile(language, text):
+    if not isinstance(language, str) or not isinstance(text, str):
+        return None
+    dictionary = calculate_frequencies(tokenize(text))
+    language_profile = {}
+    language_profile['name'] = language
+    language_profile['freq'] = dictionary
+    #print(language_profile)
+    return language_profile
+
+
+def calculate_mse(predicted, actual):
+    if not isinstance(predicted, list) or not isinstance(actual, list):
+        return None
+    if isinstance(predicted, list):
+        for i in predicted:
+            if not isinstance(i, (float, int)):
+                return None
+            else:
+                continue
+        if isinstance(actual, list):
+            for i in actual:
+                if not isinstance(i, (float, int)):
+                    return None
+                else:
+                    continue
+    if len(predicted) != len(actual):
+        return None
+    error_score_list = []
+    for i in range(len(predicted)):
+        error_score = (predicted[i] - actual[i])**2
+        error_score_list.append(error_score)
+    mse = sum(error_score_list)/len(actual)
+    return mse
+
+
+def compare_profiles(unknown_profile, profile_to_compare):
+    # bad input check for unknown_profile:
+    if not isinstance(unknown_profile, dict):
+        return None
+    if isinstance(unknown_profile, dict):
+        for i in unknown_profile['freq'].keys():
+            if not isinstance(i, str):
+                return None
+            else:
+                continue
+        for i in unknown_profile['freq'].values():
+            if not isinstance(i, (int, float)):
+                return None
+            else:
+                continue
+
+    # bad input check for profile_to_compare:
+    if not isinstance(profile_to_compare, dict):
+        return None
+    if isinstance(profile_to_compare, dict):
+        for i in profile_to_compare['freq'].keys():
+            if not isinstance(i, str):
+                return None
+            else:
+                continue
+        for i in profile_to_compare['freq'].values():
+            if not isinstance(i, (int, float)):
+                return None
+            else:
+                continue
+    if len(profile_to_compare) == 2 and len(unknown_profile) == 2:
+        freq_dict_2 = profile_to_compare['freq']
+        keys_of_dict_2 = freq_dict_2.keys()
```

Comment:
use the keys() method in a loop, do not put it in a separate variable

Reply:
done